admin/users/views.py

```diff
@@ -456,10 +457,10 @@ def get_context_data(self, **kwargs):
 
 class GetUserConfirmationLink(GetUserLink):
     def get_link(self, user):
-        try:
-            return user.get_confirmation_url(user.username, force=True)
```

Comment:
[continued] I now realize that this call was supposed to already be exhibiting this behavior (i.e. creating/refreshing tokens), and that the real issue is that this `force` flag doesn't work in all cases.

The best approach to fixing this bug might actually be to resolve [that `TODO` from 2016](https://github.com/CenterForOpenScience/osf.io/blob/63365d8e73936c27c258a150c9043f349b00e599/osf/models/user.py#L1300), by adding something like this to the OSFUser model

```python
def get_or_create_confirmation_url(self, email, force=False, renew=False)
    # May raise ValidationError for banned email domains
    try:
        self.get_confirmation_token(email, force=force, renew=renew)
    except KeyError:
        self.add_unconfirmed_email(email)
        self.save()
    return self.get_confirmation_url(email)
```
and call that here. That _should_ cover the expected failure cases from `get_confirmation` -- no existing link, expired, or no existing email. The confirmed logic is already handled in the template, but the deleted/merged logic could be handled here in `get_link`